.github/workflows/build-ampd-handlers-release.yaml

```diff
@@ -0,0 +1,266 @@
+    needs: extract-semver
+    permissions:
+      contents: write
+      packages: write
+      id-token: write
+
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@7474bc4690e29a8392af63c5b98e7449536d5c3a # v4
+        with:
+          aws-region: us-east-2
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/ghwf-${{ github.event.repository.name }}
+
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+        with:
+          fetch-depth: '0'
+          ref: ${{ github.event.inputs.tag }}
+          submodules: recursive
+
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+        with:
+          cache-key: "cache-release-ubuntu-22.04-amd64"
+
+      - name: Import GPG key
+        id: import_gpg
+        uses: crazy-max/ghaction-import-gpg@e89d40939c28e39f97cf32126055eeae86ba74ec # v6
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+
+      - name: build and sign linux binary
+        env:
+          SEMVER: ${{ needs.extract-semver.outputs.semver }}
+          HANDLER: ${{ needs.extract-semver.outputs.handler }}
+        run: |
+          OS="linux"
+          ARCH="amd64"
+          sudo apt-get update
+          sudo apt-get install -y protobuf-compiler
+
+          cargo build --release --bin ${HANDLER}-handler
+          if [ $? -ne 0 ]; then
+            echo "Error: cargo build failed"
+            exit 1
+          fi
+          
+          mkdir handlerbin
+          mv "/home/runner/work/axelar-amplifier/axelar-amplifier/target/release/${HANDLER}-handler" "./handlerbin/${HANDLER}-handler-$OS-$ARCH-$SEMVER"
+
+          gpg --armor --detach-sign "./handlerbin/${HANDLER}-handler-$OS-$ARCH-$SEMVER"
+
+      - name: Test Binary Format
+        working-directory: ./handlerbin
+        env:
+          HANDLER: ${{ needs.extract-semver.outputs.handler }}
+        run: |
+          for binary in ./${HANDLER}-handler-*; do
+          if [[ "$binary" != *.asc ]]; then
+            echo "Testing binary: $binary"
+            OUTPUT=$(file "$binary" | cut -d: -f2- | awk -F, '{print $1"," $2}')
+            EXPECTED="ELF 64-bit LSB pie executable, x86-64"
+
+            echo "Output: $OUTPUT"
+            echo "Expected: $EXPECTED"
+
+            if [[ "$OUTPUT" == *"$EXPECTED"* ]]; then
+              echo "The binary format is correct."
+            else
+              echo "Error: The binary format does not match the expected format."
+              exit 1
+            fi
+          fi
+          done
+
+      - name: Create zip and sha256 files
+        working-directory: ./handlerbin
+        run: |
+          for i in `ls | grep -v .asc`
+          do
+            shasum -a 256 $i | awk '{print $1}' > $i.sha256
+            zip $i.zip $i
+            shasum -a 256 $i.zip | awk '{print $1}' > $i.zip.sha256
```

Comment:
I think we just need the checksum hash of the actual file, checksum hash of zipped file is redundant.